File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 28.11.2025, 20:03:25
1class Solution:
2    def isAnagram(self, s: str, t: str) -> bool:
3        if len(s) != len(t): return False
4        letters = {}
5        for letter in s:
6            if letter in letters.keys():
7                letters[letter] +=1
8            else:
9                letters[letter] = 1
10        for letter in t:
11            if letter in letters.keys():
12                letters[letter] -= 1
13                if letters[letter] == -1:
14                    return False
15            else:
16                return False
17        return True

File: 242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 21.02.2026, 19:23:45
1using namespace std;
2class Solution {
3public:
4    bool isAnagram(string s, string t) {
5        int tab[26];         
6        for (int i = 0; i < s.size() ; i++){
7            tab[((int) s[i])%26] += 1;
8        }
9        for (int i = 0; i < t.size() ; i++){
10            tab[((int) t[i])%26] -= 1;
11        }
12        for (int elem : tab){
13            if (elem != 0){
14                return false;
15            }
16        }
17        return true;
18    }
19};